numpy/core/tests/test_multiarray.py

```diff
@@ -1073,34 +1073,60 @@ def test_array_too_big(self):
             assert_raises(ValueError, np.ndarray, buffer=buf, strides=(0,),
                           shape=(max_bytes//itemsize + 1,), dtype=dtype)
 
-    def test_jagged_ndim_object(self):
```

Comment:
These are the real NEP 34 tests (from here down in this file): they test that the behaviour has not changed when using the `dtype=object` kwarg, and that the warning does not seem to have false-positives when no dtype is used.